scripts/db_tools/import-project.ts

```diff
@@ -0,0 +1,258 @@
+        type: 'string',
+        choices: ['dev', 'qa', 'live'],
+        default: 'live',
+        requiresArg: true,
+        description: 'machine to fetch mongodb data from'
+      })
+      .option('destinationMachine', {
+        type: 'string',
+        choices: ['dev', 'qa', 'live'],
+        default: 'dev',
+        requiresArg: true,
+        description: 'machine to write mongodb data to'
+      })
+      .option('originProjectSFId', {
+        type: 'string',
+        demandOption: true,
+        requiresArg: true,
+        description: 'project SF id in origin DB to import'
+      })
+      .option('destinationUserSFId', {
+        type: 'string',
+        requiresArg: true,
+        description: 'Replace destination project userRoles with this user SF id as a project administrator'
+      })
+      .option('newDestinationProjectPTId', {
+        type: 'string',
+        requiresArg: true,
+        description:
+          'Set the imported destination project paratextId to this value. This will have an important impact on Synchronization and PT Registry lookups'
+      })
+      .strict()
+      .parseSync();
+
+    const shouldUseColor: boolean = args.color;
+    useColor(shouldUseColor);
+    this.originMachine = args.originMachine;
+    this.destinationMachine = args.destinationMachine;
+    this.originConnectionConfig = databaseConfigs.get(this.originMachine);
+    this.destinationConnectionConfig = databaseConfigs.get(this.destinationMachine);
+    this.destinationUserSFId = args.destinationUserSFId;
+    this.newDestinationProjectPTId = args.newDestinationProjectPTId;
+    return args.originProjectSFId;
+  }
+
+  async withDB(
+    activity: (originConn: Connection, originDB: Db, destinationConn: Connection, destinationDB: Db) => Promise<void>
+  ): Promise<void> {
+    if (this.destinationConnectionConfig == null) {
+      throw new Error('null destination connection config');
+    }
+    if (this.originConnectionConfig == null) {
+      throw new Error('null origin connection config');
+    }
+    console.log(`Connecting to origin '${this.originMachine}'.`);
+    const originWS: WebSocket = createWS(this.originConnectionConfig);
+    const originConn: Connection = new Connection(originWS);
+    const originClient: MongoClient = await MongoClient.connect(this.originConnectionConfig.dbLocation);
+    console.log(`Connecting to destination '${this.destinationMachine}'.`);
+    const destinationWS: WebSocket = createWS(this.destinationConnectionConfig);
+    const destinationConn: Connection = new Connection(destinationWS);
+    const destinationClient: MongoClient = await MongoClient.connect(this.destinationConnectionConfig.dbLocation);
+    try {
+      const originDB: Db = originClient.db();
+      const destinationDB: Db = destinationClient.db();
+      await activity(originConn, originDB, destinationConn, destinationDB);
+    } finally {
+      await originClient.close();
+      (originConn as any).close();
```

Comment:
I think this will create a button to "commit suggestion":
``` diff
-      (originConn as any).close();
+      originWS.close();
       await destinationClient.close();
-      (destinationConn as any).close();
+      destinationWS.close();
```

Reply:
Nope. I have to write it like this:
````
``` suggestion
diff
```
````
I'll try again on Reviewable.

Reply:
I don't think it will work on Reviewable. Trying it on GitHub.
``` suggestion
-      (originConn as any).close();
+      originWS.close();
       await destinationClient.close();
-      (destinationConn as any).close();
+      destinationWS.close();
```